modules/fun.js

```diff
@@ -0,0 +1,450 @@
+    const random = Math.floor(Math.random() * possibleSpaces.length);
+    mineSpaces.push(possibleSpaces[random]);
+    possibleSpaces.splice(random, 1);
+  }
+
+  function getMineCount(x, y) {
+    let count = 0;
+    for (let i = -1; i <= 1; i++) {
+      if ((x + i) < 0 || (x + i) >= size) continue;
+      for (let j = -1; j <= 1; j++) {
+        if ((y + j) < 0 || (y + j) >= size) continue;
+        if (mineSpaces.includes((y + j) * size + x + i)) count++;
+      }
+    }
+
+    return count;
+  }
+
+  // Creating the final board
+  /** @type {number[][]} */
+  const board = [];
+  for (let x = 0; x < size; x++) {
+    board.push([]);
+    for (let y = 0; y < size; y++) {
+      if (mineSpaces.includes(x + (y * size))) {
+        board[x].push(9);
+        continue;
+      }
+      board[x].push(getMineCount(x, y));
+    }
+  }
+  const output = board.map(row => row.map(num => `||${num == 9 ? mineSweeperEmojis[`bomb`] : mineSweeperEmojis[num]}||`).join(``)).join(`\n`);
+  field = (`**Mines: ${mineCount}** (Tip: Corners are never mines)\n${output}`);
+  // we need to split it up because only 99 emoji per message limit for some reason.
+  let degradingField = field;
+  function countEmoji(text) {
+    const emojiRegex = new RegExp(`(${Object.values(mineSweeperEmojis).map(emoji => emoji.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')).join(`|`)})`, 'g');
+    const emoji = text.match(emojiRegex);
+    return emoji?.length || 0;
+  }
+  if (countEmoji(field) <= 99) {
+    return int.editReply(field); // No splitting needed
+  }
+  while (countEmoji(degradingField) > 99) {
+    let segment = ``;
+    while (countEmoji(segment + degradingField.substring(0, degradingField.indexOf(`\n`) >= 0 ? degradingField.indexOf(`\n`) : degradingField.length)) <= 99) {
+      segment += degradingField.substring(0, (degradingField.indexOf(`\n`) >= 0 ? degradingField.indexOf(`\n`) : degradingField.length) + 1);
+      degradingField = degradingField.substring((degradingField.indexOf(`\n`) >= 0 ? degradingField.indexOf(`\n`) : degradingField.length) + 1);
+    }
+    if (segment + degradingField == field) {
+      await int.editReply(segment);
+    } else {
+      await int.channel.send(segment);
+    }
+  }
+  return int.channel.send(degradingField);
+}
+/**
+ * function slashFunRoll
+ * @param {Discord.ChatInputCommandInteraction} int a /fun roll interaction
+ */
+async function slashFunRoll(int) {
+  let dice = int.options.getInteger('dice');
+  let sides = int.options.getInteger('sides');
+  let modifier = int.options.getInteger('modifier');
+  if (!dice) dice = 1;
+  if (!sides) sides = 6;
+  if (!modifier) modifier = 0;
+  /** @type {string[][]} */
+  const rolls = [];
+  let total = 0;
+  rolls[sides] = [];
+  const num = dice;
+  if (num && num <= 10000) {
+    for (let i = 0; i < num; i++) {
+      const val = Math.ceil(Math.random() * sides);
+      rolls[sides].push((i == 0 ? `**d${sides}:** ` : ``) + val);
+      total += val;
+    }
```

Comment:
We're not dealing with multiple types of dice, so we can reduce this to a regular number array
```suggestion
  /** @type {number[]} */
  const rolls = [];
  for (let i = 0; i < dice; i++) {
    rolls.push(Math.ceil(Math.random() * sides);
}

Reply:
I still like the look of the old one's returns, I reorganized and cut out and reformed a bunch of this code.

Reply:
so this review isn't accurate much anymore...